Automattic/wp-calypso
# Refactor Jetpack CRO pricing page iterations

#### Changes proposed in this Pull Request

* Replace `getJetpackCROActiveVersion` with new convenience functions in `iterations.ts`. In these new functions, the active iteration is resolved "behind-the-scenes" and passed in as a parameter.
* Attempt to reduce the overall surface area of code that's aware of the current CRO iteration.
* Move iteration-specific components into sub-directories labeled with the iteration they belong to.

#### Testing instructions

**TIP:** This is a very large PR in terms of lines of code. If it's too much to digest at once, try viewing each individual commit on its own to make it easier. 🙂 

1. **Thorough code review.** This PR is aimed primarily at making development easier, so if it doesn't accomplish that somewhere, call it out! Feedback is very much encouraged, no matter how large or small. As noted below, pay special attention to `iterations.ts`.
2. **Full regression testing.** Explore all pricing page, product, and plan-related functions in Calypso Blue, Green, and Jetpack Connect. Ensure that behavior is as you would expect and matches the current production environments on WordPress.com and Jetpack.com. Switch between the two active iterations during testing to make sure both still function as intended.

Related to #49474, `pbtFFM-AH-p2`.
